### DISCUSSION TITLE
Erreur lecture de type (int64 / int)

### DISCUSSION THREAD

**Discussion by C4rdiaK** [danymat/INSAnonym-utils#82] [Q&A, Bug report & Support] [answered]
Bonjour,

Lors de la lecture d'une base de donnée, les données au format int (comme l'id d'un utilisateur par exemple) nous sont retournées sous le format int64, ce qui engendre l'erreur suivante:
` raise Exception(f"Error in column types: column {column.name} if of type {type}, not {column.column_type}") `
`Exception: Error in column types: column id if of type int64, not int`
Ce problème peut être contourné en commentant la vérification des types dans la librairie, ce qui est moins qu'optimal (et barbare).

**Comment by danymat** [comment, marked as answer]
Bonjour, vous êtes sur quelle version de python ?

**Reply by beng-git** [reply]
Bonjour, est-ce que ce problème a été résolu ?

**Reply by danymat** [reply]
Bonjour, je n'ai reçu aucun nouveau message, je ne peux donc pas reproduire le bug. Je pense qu'ils ont réussi à corriger le bug

**Reply by C4rdiaK** [reply]
Bonjour, 
désolé pour l'absence de réponse. Ma petite solution barbare ayant fonctionné j'ai oublié jusqu'a l'existence de ce bug et aussi de ce thread.
Je n'ai donc pas réussi à trouver de manière propre de résoudre le problème.
Je travaille sur Python3.9.8

**Reply by YeloDriver** [reply]
Bonjour, nous avons trouvé le même problème, on a testé sur python 3.9.8 / 3.8.9 / 3.8.10

**Reply by beng-git** [reply]
En testant avec un autre groupe sur un problème qui a l'air différent mais se matérialise également par un soucis de typage, ils ont des erreurs avec les versions plus récentes de python, pandas et numpy, par contre moi avec la version 3.7.1 de python, 1.15.0 de numpy et 0.23.4 de pandas je n'ai pas d'erreur. Une chose que vous pouvez donc tester, avant de voir si on peut patcher, c'est de downgrader à la version 3.7 de python.

**Reply by beng-git** [reply]
En testant un peu plus avec la version 3.9.2, numpy 1.20.1 et pandas 1.3.4 j'ai le même soucis.

Le patch que je propose est le suivant : fichier utility_POI.py remplacer dans les lignes 65 et 66 int64 par str :  
```
df = df.astype({'longitude': 'float64', 'latitude': 'float64', 'id': 'str'})
df_orig = df_orig.astype({'longitude': 'float64', 'latitude': 'float64', 'id': 'str'})
```

Même chose pour les lignes 10 et 11 de utility_meet.py et 10 et 11 de utility_tuile.py

   Cordialement,

BN

**Reply by danymat** [reply]
Bonsoir, j'ai autorisé dans le modèle l'utilisation de `int64` à la place de `int`.

Essayez de modifier les références de `int` vers `int64` dans les colonnes qui posent problème.

```py
 "columns": [
    ...
    { "name": "id", "column_type": "int64" } , # modifiez int vers int64
   ...
```

Bien sûr, faites `pip install insanonym-utils --upgrade` avant de tester

**Reply by YeloDriver** [reply]
Yes, il me semble ce problème est résolu

**Reply by danymat** [reply]
Super, @C4rdiaK si le problème est réglé, tu peux régler cette discussion comme résolue.
